tests/integration/test_route53resolver.py

```diff
@@ -0,0 +1,570 @@
+    @pytest.mark.skip_snapshot_verify(paths=["$..SecurityGroupIds"])
+    def test_create_resolver_rule_with_invalid_direction(
+        self, ec2_client, route53resolver_client, cleanups, snapshot
+    ):
+        request_id = short_uid()
+        resolver_endpoint_name = f"rs-{request_id}"
+        result = route53resolver_client.create_resolver_endpoint(
+            CreatorRequestId=request_id,
+            SecurityGroupIds=self.get_security_group_ids(ec2_client),
+            Direction="INBOUND",
+            Name=resolver_endpoint_name,
+            IpAddresses=self.get_create_resolver_endpoint_ip_address(ec2_client, "28"),
+        )
+
+        create_resolver_endpoint_res = result.get("ResolverEndpoint")
+
+        if self._wait_created_endpoint_is_listed_with_status(
+            route53resolver_client, request_id, "OPERATIONAL"
+        ):
+            create_resolver_endpoint_res["Status"] = "OPERATIONAL"
+
+        snapshot.match("create_resolver_endpoint_res", create_resolver_endpoint_res)
+
+        with pytest.raises(Exception) as inavlid_request:
+            route53resolver_client.create_resolver_rule(
+                CreatorRequestId=short_uid(),
+                RuleType="FORWARD",
+                DomainName="www.example.com",
+                ResolverEndpointId=create_resolver_endpoint_res["Id"],
+                TargetIps=[
+                    {"Ip": "10.0.1.200", "Port": 123},
+                ],
+            )
+
+        msg = "Resolver rules can only be associated to OUTBOUND resolver endpoints."
+        snapshot.add_transformer(snapshot.transform.key_value("Message", msg))
+        snapshot.add_transformer(snapshot.transform.jsonpath("$..Error.Message", msg))
+        snapshot.match("invalid_request_ex", inavlid_request.value.response)
+
+        # clean up
+        cleanups.append(
+            lambda: route53resolver_client.delete_resolver_endpoint(
+                ResolverEndpointId=create_resolver_endpoint_res["Id"]
+            )
+        )
+
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify(paths=["$..SecurityGroupIds", "$..ShareStatus"])
+    def test_multipe_create_resolver_rule(
+        self, ec2_client, route53resolver_client, cleanups, snapshot
+    ):
+        request_id = short_uid()
+        resolver_endpoint_name = f"rs-{request_id}"
+        result = route53resolver_client.create_resolver_endpoint(
+            CreatorRequestId=request_id,
+            SecurityGroupIds=self.get_security_group_ids(ec2_client),
+            Direction="OUTBOUND",
+            Name=resolver_endpoint_name,
+            IpAddresses=self.get_create_resolver_endpoint_ip_address(ec2_client, "18"),
+        )
+
+        create_resolver_endpoint_res = result.get("ResolverEndpoint")
+
+        if self._wait_created_endpoint_is_listed_with_status(
+            route53resolver_client, request_id, "OPERATIONAL"
+        ):
+            create_resolver_endpoint_res["Status"] = "OPERATIONAL"
```

Comment:
why is this patching the result?

Reply:
_**client.create_resolver_endpoint**_ gives the status of the endpoint as **CREATING**, takes about 1min for inbound and 2-3min for outbound endpoint to create and change the status to **OPERATIONAL** while listing the endpoints. 
Whereas we run this locally, _**client.create_resolver_endpoint**_  states "OPERATIONAL" while listing the endpoints immediately.

Reply:
Can you please tell me more about the approach?

Reply:
Something like this:

```python
def create_resolver_endpoint(....):
    ...
    resolver_endpoint = {
        ...
        "Status": "OPERATIONAL"
    }
    store[someid] = resolver_endpoint.copy()
    resolver_endpoint["Status"] = "CREATING
    return resolver_endpoint
```


This assumes that was you've said on localstack the created resource is immediately available while on AWS it always takes at least *some* additional time to provision.

Reply:
got it